lib/icingadb/icingadb.cpp

```diff
@@ -67,8 +69,23 @@ void IcingaDB::Start(bool runtimeCreated)
 	m_Rcon = new RedisConnection(GetHost(), GetPort(), GetPath(), GetPassword(), GetDbIndex(),
 		GetEnableTls(), GetInsecureNoverify(), GetCertPath(), GetKeyPath(), GetCaPath(), GetCrlPath(),
 		GetTlsProtocolmin(), GetCipherList(), GetDebugInfo());
-	m_Rcon->SetConnectedCallback([this](boost::asio::yield_context& yc) {
-		m_WorkQueue.Enqueue([this]() { OnConnectedHandler(); });
+
+	m_Rcon->SetConnectedCallback([this](yield_context& yc) {
+		m_Rcon->SetConnectedCallback({});
+		m_RconsPending.store(m_Rcons.size());
+
+		for (auto& kv : m_Rcons) {
```

Comment:
`m_Rcons` is only initialized after this connection is started, so this access could happen while that map is not yet fully initialized. Simply moving setting this callback and the call fo `Start()` after the loop should fix this.

Reply:
(This was already introduced by #8933, somehow I've missed that there.)

Reply:
That’s #8932's purpose.